hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/throttling/FunctionalityThrottling.java

```diff
@@ -54,6 +54,19 @@ default boolean shouldThrottleNOfUnscaled(final int n, final HederaFunctionality
 
     List<DeterministicThrottle> allActiveThrottles();
 
+    /**
+     * Returns a list of snapshots of the current usage of all active throttles.
+     * @return the active snapshots
+     */
+    List<DeterministicThrottle.UsageSnapshot> takeSnapshots();
```

Comment:
Oh but wait, here and in the method decl below you _do_ have the opportunity to specify `@NonNull` return and argument.